### PR TITLE
During PayPal express checkout all gateways are available as a payment method (850)

### PR DESCRIPTION
**Pull request by Dinamiko**
When customer perform express checkout, on a checkout page all gateways are available as payment options.

### Steps to Test

- Enable all gateways
- Place the order with PayPal from mini cart, product or cart page

### What should happen
When redirected to checkout page customer only should see PayPal gateway.

### What happened instead
When redirected to checkout page customer will see all gateways as an available payment method.

### CODE REVIEW

File: modules/ppcp-wc-gateway/src/Checkout/DisableGateways.php [hunk 1]
```diff
@@ -107,7 +107,7 @@ private function disable_all_gateways() : bool {
 	 */
 	private function needs_to_disable_gateways(): bool {
 		$wc_ajax = filter_input( INPUT_GET, 'wc-ajax', FILTER_SANITIZE_STRING ) ?? '';
-		if ( $wc_ajax === 'update_order_review' ) {
+		if ( $wc_ajax === 'update_order_review' && $this->session_handler->funding_source() !== 'paypal' ) {
```
Comment (AlexP11223) [inline modules/ppcp-wc-gateway/src/Checkout/DisableGateways.php]: hm, will it still work correctly when using other buttons like Venmo, Pay Later?

Comment (Dinamiko) [reply modules/ppcp-wc-gateway/src/Checkout/DisableGateways.php:110]: @AlexP11223 yup it wirks for other buttons, the fix was for not hidding other gateways after getting a WC validation error when using DCC payment. I used `update_order_review` to ensure we are on that context, but when coming from express checkout `update_order_review` also exist and hides the other gateways, that´s the reason I added the conditional to not execute it when is `paypal`.

Comment (Dinamiko) [reply modules/ppcp-wc-gateway/src/Checkout/DisableGateways.php]: you´re right, it only handles PayPal button.

**Comment by Dinamiko** [comment]
@AlexP11223 I´ve just noticed the original issue was fixed here: https://github.com/woocommerce/woocommerce-paypal-payments/commit/c11635343d1c780d1f1ffe7fc0b7d74a8bdfbec5#diff-0999d0e241de01c000b49cd2a486a81fb1d4a51614fa78cea314b7e7fddc7538R114-R116

So no need for checking `update_order_review` here.

**Review by AlexP11223 — APPROVED** [review]

**Pull request merged**